geopandas/geodataframe.py

```diff
@@ -441,6 +441,53 @@ def plot(self, *args, **kwargs):
     plot.__doc__ = plot_dataframe.__doc__
 
 
+    def dissolve(self, by=None, aggfunc='first'):
+        """
+        Dissolve geometries within `groupby` into single observation. 
+
+        Parameters
+        ----------
+        by : string, default None
+            Column whose values define groups to be dissolved
+        aggfunc : function or string, default "first"
+            Aggregation function for manipulation of data associated
+            with each group. Passed to pandas `groupby.agg` method.
+
+        Returns
+        -------
+        GeoDataFrame
+        """
+
+        # Process non-spatial component
+        data = self.drop(labels=self.geometry.name, axis=1).copy()
+        aggregated_data = data.groupby(by=by).agg(aggfunc)
```

Comment:
Do we want `as_index=False` here? Or at least have the possibility to pass this through?